js/components/frame.js

```diff
@@ -802,7 +810,11 @@ class Frame extends ImmutableComponent {
       const isError = this.props.aboutDetails && this.props.aboutDetails.get('errorCode')
       if (!this.props.isPrivate && this.props.provisionalLocation === this.props.location && (protocol === 'http:' || protocol === 'https:') && !isError && savePage) {
         // Register the site for recent history for navigation bar
-        appActions.addSite(siteUtil.getDetailFromFrame(this.frame))
+        const addSiteDebounced = debounce((frame) => {
```

Comment:
The problem is sometimes we get a loadi-end but the favicon or themecolor event didn't happen yet.  So you need to also call addSiteDebounced from:
 `this.webview.addEventListener('page-favicon-updated', (e) => {`
`this.webview.addEventListener('did-change-theme-color', ({themeColor}) => {`

But be sure not to call it for private tabs, like the above one doesn't.